sketches/BLE_Scratch/BLE_Scratch.ino

```diff
@@ -495,3 +505,50 @@ void setLedPinValue(int pin, int value) {
     analogWrite(pin, 255 - value);
   }
 }
+
+enum robotAction {
+  MOVE_FORWARD_STEP = 0,
+  MOVE_BACKWARD_STEP = 1,
+  TURN_RIGHT = 2,
+  TURN_LEFT = 3,
+  SET_SPEED = 4,
+  MOVE_FORWARD_TIME = 5,
+  MOVE_BACKWARD_TIME = 6,
+};
+
+void onRobotActionCharacteristicWrite(BLEDevice central, BLECharacteristic characteristic) {
+  enum pinAction action = (enum pinAction)pinRobotCharacteristic[0];
```

Comment:
in cpp, you don't need to use write again `enum`.  I will also use the more idiomatic `static_cast`
```suggestion
  pinAction action = static_cast<pinAction>(pinRobotCharacteristic[0]);
```